File: spec/php/StrEncodingsDefaultTest.php
<?php
namespace Kaitai\Struct\Tests;

class StrEncodingsDefaultTest extends TestCase {
    public function testStrEncodingsDefault() {
        $r = StrEncodingsDefault::fromFile(self::SRC_DIR_PATH . "/str_encodings.bin");

        $this->assertEquals('Some ASCII', $r->str1);
        $this->assertEquals('こんにちは', $r->rest->str2);
        $this->assertEquals('こんにちは', $r->rest->str3);
        $this->assertEquals('░▒▓', $r->rest->str4);
    }
}

File: spec/go/str_encodings_default_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestStrEncodingsDefault(t *testing.T) {
	f, err := os.Open("../../src/str_encodings.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r StrEncodingsDefault
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, "Some ASCII", r.Str1)
	assert.EqualValues(t, "こんにちは", r.Rest.Str2)
	assert.EqualValues(t, "こんにちは", r.Rest.Str3)
	assert.EqualValues(t, "░▒▓", r.Rest.Str4)
}
